java/src/org/openqa/selenium/remote/http/RetryRequest.java

```diff
@@ -30,16 +30,62 @@
 import dev.failsafe.Fallback;
 import dev.failsafe.RetryPolicy;
 import java.net.ConnectException;
-import java.util.concurrent.atomic.AtomicReference;
 import java.util.logging.Logger;
 import org.openqa.selenium.TimeoutException;
 
 public class RetryRequest implements Filter {
 
   private static final Logger LOG = Logger.getLogger(RetryRequest.class.getName());
-  private static final AtomicReference<HttpResponse> fallBackResponse = new AtomicReference<>();
 
-  private static final Fallback<Object> fallback = Fallback.of(fallBackResponse::get);
+  private static final Fallback<Object> fallback =
+      Fallback.of(
+          executionAttemptedEvent -> {
+            if (executionAttemptedEvent.getLastException() != null) {
```

Comment:
Given the length of this, I'd suggest pulling out it out into a `private` method in this class. It'll make it easier to visually figure out the fields of the class from the methods it contains.

Reply:
Ahh makes sense. Thank you! Will do that.